nixos/modules/services/networking/networkmanager.nix

```diff
@@ -325,16 +318,75 @@ in {
         '';
       };
 
+      enableFortiSSL = mkOption {
```

Comment:
I think we might even just remove the VPN plug-ins from basePackages and let users install the VPN providers manually to `netowrk-manager.packages`. And using `passthru` to let users know what files to link to `environment.etc` and `tmpfiles.d` (I think there should be a way to make these unnecessary but we can leave that for the future.)

Reply:
In the current state this PR already removes all VPN plugins from the (now-gone) `basePackages`. they get added and configured conditionally on their respective options.

I think this is superior, even if more repetitive on our end, then making the user fiddle with `passthru`.

Reply:
Maintainer would set the `passthru` for each plug-in and user would just use `netowrk-manager.packages = [ nm-foovpn … ];`.

Reply:
Re: tmpfiles.d https://github.com/NixOS/nixpkgs/issues/40594

Reply:
I kinda like that @jtojnar. But the thing here with making them all optional, they're currently all off by default and that's going to break existing setups that might have relied on it. So we'd have to add a default set to `network-manager.packages`, and I guess some sort of `excludedPackages` thing, with all the default plugins in `relatedPackages` so users know which one's are there default.  

I think having the plugins in a list would make sense.
Should we name it `network-manager.plugins`?

But if we decide to have an option per plugin, we need to have all the plugins that were in basePackages to be default `true`.

Reply:
I do not see a problem with changing defaults, as long as it is properly documented in release notes.

Reply:
If it doesn't get too verbose to use, I'd also prefer having a list like `network-manager.[packages|plugins]` over adding tons of options.

:+1: on @jtojnar - we can ship less plugins by defaults, if it's properly covered in the release notes.

Reply:
I disapprove of this because it isn't a seamless transition. Usually if someone has something installed or configured they'd want this to stay that way. Not to mention we'd probably cause users headache's on nixos-unstable.

This PR will also have to re-add the disabled plugins for many of the desktop manager's that make inferences that these plugins are already there.

Reply:
But as mentioned by @jtojnar, we have to add information to the respective packages in `passthru` that is done intricately here already.

Reply:
I'm not so sure about these two comments.

 - I think doing these changes is fine if they're documented in the release notes.
 - Desktop Environments like plasma and gnome shell could add them (and even if they wouldn't, I think they kinda enumerate the available plugins installed)